packages/domain-picker/test/suggestion-item.tsx

```diff
@@ -15,293 +16,76 @@ import { render, fireEvent, screen } from '@testing-library/react';
-
-			recordAnalytics.mockClear();
+			expect( MOCK_PROPS.onRender ).toHaveBeenCalledTimes( 1 );
+		} );
 
-			rerender( <SuggestionItem { ...props } /> );
+		it( 'should not send a render event when re-rendered with the same props', async () => {
+			const { rerender } = renderComponent();
 
-			expect( recordAnalytics ).not.toHaveBeenCalled();
+			rerender( <SuggestionItem { ...MOCK_PROPS } /> );
+			expect( MOCK_PROPS.onRender ).toHaveBeenCalledTimes( 1 );
 		} );
 
-		it( "doesn't send render event when unrelated props change", async () => {
-			// Delay import so we have time to load configData in `beforeAll`
-			const { default: SuggestionItem } = await import( '../suggestion-item' );
-
-			const unchangedProps = {
-				isRecommended: true,
-				railcarId: 'id',
-				uiPosition: 113,
-			};
-
-			const { rerender } = render(
-				<SuggestionItem
-					{ ...unchangedProps }
-					suggestion={ { ...testSuggestion, domain_name: 'example.com' } }
-					isSelected={ false }
-					onSelect={ jest.fn() }
-					recordAnalytics={ jest.fn() }
-				/>
-			);
-
-			const recordAnalytics = jest.fn();
-
-			rerender(
-				<SuggestionItem
-					{ ...unchangedProps }
-					// Suggestion object is changed, but the domain name itself isn't changed
-					suggestion={ { ...testSuggestion, domain_name: 'example.com' } }
-					isSelected={ true }
-					onSelect={ jest.fn() }
-					recordAnalytics={ recordAnalytics }
-				/>
-			);
-
-			expect( recordAnalytics ).not.toHaveBeenCalled();
-		} );
+		it( 'should not send a render event when unrelated props change', async () => {
+			const { rerender } = renderComponent();
+			const updatedProps = { ...MOCK_PROPS, cost: '€11' };
+			rerender( <SuggestionItem { ...updatedProps } /> );
 
-		it( 'sends render event when domain name and railcarId changes', async () => {
-			// Delay import so we have time to load configData in `beforeAll`
-			const { default: SuggestionItem } = await import( '../suggestion-item' );
-
-			const recordAnalytics = jest.fn();
-
-			const props = {
-				isSelected: false,
-				onSelect: jest.fn(),
-				railcarId: 'id1',
-				recordAnalytics,
-			};
-
-			const { rerender } = render(
-				<SuggestionItem
-					{ ...props }
-					suggestion={ { ...testSuggestion, domain_name: 'example1.com' } }
-					isRecommended={ true }
-					railcarId="id1"
-					uiPosition={ 1 }
-				/>
-			);
-
-			expect( recordAnalytics ).toHaveBeenCalled();
-
-			rerender(
-				<SuggestionItem
-					{ ...props }
-					suggestion={ { ...testSuggestion, domain_name: 'example2.com' } }
-					isRecommended={ false }
-					railcarId="id2"
-					uiPosition={ 2 }
-				/>
-			);
-
-			expect( recordAnalytics ).toHaveBeenCalledTimes( 2 );
-			expect( recordAnalytics ).toHaveBeenLastCalledWith(
-				expect.objectContaining( {
-					railcarId: 'id2',
-					result: 'example2.com',
-					trainTracksType: 'render',
-					uiPosition: 2,
-				} )
-			);
+			expect( updatedProps.onRender ).toHaveBeenCalledTimes( 1 );
```

Comment:
Same comment as per https://github.com/Automattic/wp-calypso/pull/51177/files#r599343262

I would clear the mock before the re-render, and check that the mock was not called.